### DISCUSSION TITLE
Need some clarification on DB charset for Craft 5

### DISCUSSION THREAD

**Discussion by JulieVelghe** [craftcms/cms#15300] [General]
Hi all,
I'm preparing some of our client sites for the upgrade to Craft 5. In the guide, I've read that you have to set the charset and collation, then update, and then remove the charset and collation settings and do a convert-charset.
I'm wondering, if you then want to put your changes live, let's say via autogit, do you then have to copy your local db back to the live environment? 
The guide only mentions the composer.json, composer.lock and config/project files (and any other template, config or modules), what about the convert-charset that we had to do locally?

**Comment by brandonkelly** [comment]
Good question. If you’re planning on upgrading to Craft 5 directly on production, you’d want to hold off on removing the overridden `charset`/`collation` settings until _after_ you’ve done that:

1. Set `CRAFT_DB_CHARSET=utf8mb3` and `CRAFT_DB_COLLATION=utf8mb3_general_ci` environment variables locally and on production
2. Upgrade to Craft 5 locally
3. Deploy to production and run `craft up` there
4. Remove the `CRAFT_DB_CHARSET` and `CRAFT_DB_COLLATION` environment variables locally and on production
5. Run `db/convert-charset` locally and on production

Alternatively, you could do everything locally and restore your local backup on production once you’re done:

1. Get a fresh database backup from production and restore it locally
2. Set `CRAFT_DB_CHARSET=utf8mb3` and `CRAFT_DB_COLLATION=utf8mb3_general_ci` environment variables locally (only)
3. Upgrade to Craft 5 locally
4. Remove the `CRAFT_DB_CHARSET` and `CRAFT_DB_COLLATION` environment variables locally
5. Run `db/convert-charset` locally
6. Create a database backup locally and restore it on production
7. Deploy to production

We’ll see if we can make this clearer in the docs.

**Reply by thupsi** [reply]
This has been a little unclear for me also. For the "direct update on production" flow, I think that when you say "Override the settings" (step 1) you mean via a config file, not via `.env` right? Cause otherwise step 6 wouldn't make sense. Unless `db/convert-charset` updates some project config file?

**Reply by brandonkelly** [reply]
Doh, forgot we’re recommending using the environment variables. Updated the instructions for that.

**Reply by darylknight** [reply]
Please could you update the docs with this? I just went through this process and needed to ask on Discord about converting the production database
